src/ddccontrol/main.c

```diff
@@ -417,7 +431,25 @@ int main(int argc, char **argv)
 		}
 		
 		if (ctrl >= 0) {
-			if (value >= 0) {
+			if (relative) {
+				unsigned short old_value, maximum;
+				int retry, result;
+
+				for (retry = RETRYS; retry || (result = ddcci_readctrl(mon, ctrl, &old_value, &maximum)) < 0; retry--)
+					;
+
+				value += old_value;
+				if (value < 0) {
+					fprintf(stderr,_("Value cannot be lower than zero! %d -> %d\n"), old_value, value);
+					relative = 0;
+				}
+				else if (value > maximum) {
+					fprintf(stderr,_("Value cannot be higher than maximum! %d / %d\n"), value, maximum);
+					relative = 0;
+					value = -1;
+				}
+			}
+			if (value >= 0 || relative) {
```

Comment:
Is `|| relative` really needed here? If `value` is valid, then it's certainly `>= 0`, and if it's invalid, then it's always `< 0`.

Reply:
But, you don't need to fix it. Whole CLI needs to be changed.

Reply:
True, I've had a feeling that there's a redundant check somewhere. `|| relative` is truly unnecessary, since we handle out-of-boundaries input in the previous block.